code/network/multiui.cpp

```diff
@@ -4776,22 +4843,27 @@ void multi_create_list_select_item(int n)
 			multi_team_reset();
 		}
 
+		// out with the old, before we start with the new
+		memset(ng->campaign_desc, 0, MULTI_MAX_DESC_LEN);
+		memset(ng->mission_desc, 0, MULTI_MAX_DESC_LEN);
+
 		switch(Multi_create_list_mode){
 		case MULTI_CREATE_SHOW_MISSIONS:		
 			// don't forget to update the info box window thingie
 			if(Net_player->flags & NETINFO_FLAG_AM_MASTER){			
 				ship_level_init();		// mwa -- 10/15/97.  Call this function to reset number of ships in mission
 				ng->max_players = mission_parse_get_multi_mission_info( ng->mission_name );				
-				
+
 				Assert(ng->max_players > 0);
 				strcpy_s(ng->title,The_mission.name);								
 
-				// set the information area text
-				multi_common_set_text(The_mission.mission_desc);
+				// set the information area text, and update the ng struct
+				strcpy(ng->mission_desc, The_mission.mission_desc);
```

Comment:
Use `strcpy_s()` here. Also, the size of `The_mission.mission_desc` is larger than the size of `ng->mission_desc` so it will be set to an empty string on overflow.